### DISCUSSION TITLE
when I refresh a page, all fetched data will delete in Nextjs

### DISCUSSION THREAD

**Discussion by hadikman** [TanStack/query#6266] [Q&A] [answered]
Hi

I have only this problem in this path:
/zones/zone?q=1

I fetch the data in Zone component, first I check when the 'q' is available and when isQueryParam is true, the fetch will start:

![Screenshot 2023-10-30 183652](https://github.com/TanStack/query/assets/71704773/f14bb215-e1da-4bf9-be76-9dde7c621c64)

And when I refresh the page these two steps will happen:

Step 1:

![screenshot-1](https://github.com/TanStack/query/assets/71704773/1d35a988-5b92-4f4a-93ed-1ec4a70fc2ed)

Step 2:

![screenshot-2](https://github.com/TanStack/query/assets/71704773/a6787294-737d-422e-b6d2-0b58f1833056)

Even after refreshing the page, I have requests to the APIs, but it doesn't cache:

![Screenshot 2023-10-30 184207](https://github.com/TanStack/query/assets/71704773/e933bf26-415d-4ca2-99a3-a32c19356487)

I must go to the other page (except /zone/[slug]) and then come back to this page, and in this way, everything is fine!

Thank you in advance

**Comment by TkDodo** [comment, marked as answer]
wild guess: https://tkdodo.eu/blog/react-query-fa-qs#2-the-queryclient-is-not-stable

**Reply by hadikman** [reply]
Thank you very much, it works 😎!

As your guide, I moved "const queryClient = new QueryClient()" outside of App component.
